### PR TITLE
Complete invite friends feature

### PR DESCRIPTION
**Pull request by jchen54**
Completed invite friends feature, but might need to add extend field later

### CODE REVIEW

File: server/routes/invite_friends.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+let db = require('../config');
+
+module.exports = function(req, res) {
+  let name = req.body.name;
+  let email = req.body.email;
+  let invitedby_id = req.user.user_id;
+
+  db.query('INSERT into invitees (name, email, invitedby_id) VALUES ($1, $2, $3)', [name, email, invitedby_id])
```
Comment (misteryeo) [inline server/routes/invite_friends.js]: Wondering if a join table would be required here if there are multiple users who invite the same friends leading to a many-to-many relationship.

File: server/server.js [hunk 2]
```diff
@@ -31,6 +31,10 @@ app.get('/users', handler.getUsers);
 
 app.post('/events/users', passport.authenticate('facebook-token'), handler.addUsersEvents);
 
+app.get('/invite', handler.getInvitees);
+
+app.post('/invite', handler.inviteFriends);
```
Comment (misteryeo) [inline server/server.js]: Would there be a reason why we aren't going through passport authentication for this route?

File: src/client/app/CreateInviteButton.jsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    this.handleAddMoreInvitees = this.handleAddMoreInvitees.bind(this);
+  }
+
+  // componentDidMount () {
+  //
+  // }
+
+  showModal () {
+    this.refs.modal.show();
+    $.ajax({
```
Comment (misteryeo) [inline src/client/app/CreateInviteButton.jsx]: For readability, I would suggest creating a method for 'getInvitees' or something similar.

File: src/client/app/CreateInviteButton.jsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+      url: '/invite',
+      method: 'POST',
+      contentType: 'application/json',
+      data: JSON.stringify({
+        name: this.state.name,
+        email: this.state.email
+      }),
+      success: () => {
+        console.log('POST request for invite is sending');
+        this.hideModal();
```
Comment (misteryeo) [inline src/client/app/CreateInviteButton.jsx]: From a user experience perspective, it's probably better to not hide the modal each time someone submits. Perhaps we could create a separate button for 'Finished' or 'Done' to hide the modal.

File: src/client/app/CreateInviteButton.jsx (hunk 3 above)
Comment (iccir919) [inline src/client/app/CreateInviteButton.jsx]: adding on to this, if you wanted to re-render the data with the modal open then you would want to move this ajax request to another function

File: src/client/app/CreateInviteButton.jsx [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        // this.setState({inviteeList: this.state.inviteeList.push({name: this.state.name, email: this.state.email})})
+      },
+      error: (err) => {
+        console.log(err);
+      }
+    })
+  }
+
+  handleAddMoreInvitees (event) {
+    event.preventDefault();
```
Comment (iccir919) [inline src/client/app/CreateInviteButton.jsx]: what purpose does this function serve in this component?

File: src/client/app/CreateInviteButton.jsx [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+    this.setState({userWantsMoreInvitees: true});
+  }
+
+  render () {
+    return (
+      <div>
+        <button onClick={this.showModal} className="col-md-4 col-md-offset-4">Invite Friends to Join</button>
```
Comment (iccir919) [inline src/client/app/CreateInviteButton.jsx]: this button doesn't match the styling of the other buttons on the homepage

File: server/server.js (hunk 2 above)
Comment (kay-lee) [inline server/server.js]: I tried both having it and not having it for these lines, but I couldn't figure out what the difference is; anyone who can explain why they should be different?

File: server/schema.js [hunk 7]
```diff
@@ -46,5 +46,21 @@ module.exports = (db) => {
     return db.query('CREATE TABLE IF NOT EXISTS friends (\
       user1 BIGINT not null,\
       user2 BIGINT not null);')
+  })
+  .then(() => {
+    return db.query('CREATE TABLE IF NOT EXISTS invitees (\
+      invitee_id serial not null primary key,\
+      name varChar(50) unique,\
```
Comment (kay-lee) [inline server/schema.js]: I think it's nice that name and email are designed to be unique. Great!

File: server/routes/get_invitees.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+module.exports = function(req, res) {
+  let invitedby_id = req.user.user_id;
+  db.query('select * from invitees where invitedby_id=($1)', [invitedby_id])
+  .then((results) => {
+    res.send(results);
+  })
+  .catch((err) => {
+    console.log(err);
+    res.send(err);
```
Comment (iccir919) [inline server/routes/get_invitees.js]: send status codes

File: server/schema.js [hunk 9]
```diff
@@ -46,5 +46,21 @@ module.exports = (db) => {
...
+      email varChar(50) unique,\
```
Comment (iccir919) [inline server/schema.js]: maybe not unique for names

File: server/server.js (hunk 2 above)
Comment (iccir919) [inline server/server.js]: we do want it!

**Pull request closed without merging**